src/Pages/Profile/Profile.tsx

```diff
@@ -1,18 +1,38 @@
 import 'cropperjs/dist/cropper.css';
+import { localStorageManager } from "Helpers/LocalStorageManager";
 import { CurrentUserProfileHeader, PublicProfileHeader } from 'Modules/User';
 import { Component } from 'react';
 import * as React from 'react';
 import { RouteComponentProps, withRouter } from 'react-router-dom';
 import { ProfileNavBar } from './ProfileNavBar';
 import { PublicProfileNavBar } from './ProfileNavBar/PublicProfileNavBar';
 
-class ProfilePage extends Component<RouteComponentProps<any>, {}> {
+interface IStates {
+  userId: string;
+}
+
+class ProfilePage extends Component<RouteComponentProps<any>, IStates> {
   constructor(props: any) {
     super(props);
+    this.state = {
+      userId: null,
+    }
   }
 
-  public render() {
+  public componentWillMount() {
+    this.checkCurrentUser();
+  }
+
+  public checkCurrentUser() {
     const userId = this.props.match.params.userId;
+    const userInfo = localStorageManager.getUserInfo();
+    if (!userInfo || userInfo.id !== userId) {
+      this.setState({userId});
+    }
+  }
+
+  public render() {
+    const userId = this.state.userId;
```

Comment:
We should use destructuring assignment `const { userId } = this.state;` to unpack value.